### DISCUSSION TITLE
post-gazette.com: detection

### DISCUSSION THREAD

**Discussion by tstrmr** [uBlockOrigin/uAssets#25653] [General]
Sorry if this is the wrong place to post, but I've recently switched to uBO Lite due to the impending switch to MV3.  The only site I've encountered so far that's giving me a problem is `https://www.post-gazette.com/`

It detects that there's an ad blocker in use

**Comment by gorhill** [comment]
This will be fixed in next release of uBOL: <https://github.com/uBlockOrigin/uAssets/commit/a1c2dd64732dcf64de805aaede0e904485828aaf>

> Sorry if this is the wrong place to post

You can use the chat icon in the popup panel to report filtering issue on a specific site, this will give us all the information we need.

**Reply by tstrmr** [reply]
Thank you!  Sorry, I missed that icon, I'll use it in the future.

**Reply by stephenhawk8054** [reply]
Blocking `tinypass.com` will affect their subscription service `https://my.post-gazette.com/my-account/`. If I'm not wrong, these anti-adblocks are already resolved with:

https://github.com/uBlockOrigin/uAssets/blob/2dc6d5e11a28ecbc9f24701e9d9ffec62070c093/filters/filters-2022.txt#L4621-L4623

I think the filter would be available in version `2024.10.14.189`

**Reply by tstrmr** [reply]
I'm on 2024.10.14.189 now and it's still detecting uBOL.  

Is there anything I need to do to make it happen?  I've tried it on both Basic and Complete.

**Reply by stephenhawk8054** [reply]
Which mode are you using?

**Reply by tstrmr** [reply]
I've tried it with both Basic and Complete, it detects it either way.

**Reply by stephenhawk8054** [reply]
Can you test in a new browser profile with only uBOL version `2024.10.14.189` and its default settings?

**Reply by gorhill** [reply]
> If I'm not wrong, these anti-adblocks are already resolved with

You're right, I cannot reproduce with 2024.10.14.189. I could reproduce using my locally installed version but now realize it was still using locally cached outdated lists.

**Reply by tstrmr** [reply]
Still happening with a fresh profile

![image](https://github.com/user-attachments/assets/53409325-8b45-437a-a792-e09921184e0b)
![image](https://github.com/user-attachments/assets/4a28a2f5-b2db-4075-8b57-dfd9392b3658)
![image](https://github.com/user-attachments/assets/de73331f-b1aa-44fa-a047-b43f6c8a1400)

**Reply by stephenhawk8054** [reply]
What happens if you turn off uBOL?

**Reply by tstrmr** [reply]
If I turn it off, either by fully disabling the extension or turning the per-site filtering to "no filtering", the pages that require a subscirption pop up a subscription banner:
![image](https://github.com/user-attachments/assets/78ba1499-2647-4604-923f-a6ee9b3320aa)

and the ones that don't require a subscription load normally:
![image](https://github.com/user-attachments/assets/f4dd4ffb-3cc5-470b-b3d1-656b833ae745)

**Reply by stephenhawk8054** [reply]
Well, I can't reproduce the issue so I can't investigate more. The paywall modal loads normally for me.

**Reply by gorhill** [reply]
@tstrmr What lists are enabled in your uBO Lite?

---

Also, does it happen on the front page or a specific article page?

Does it happen only after browsing different number of articles?

**Reply by gorhill** [reply]
@stephenhawk8054 

I could reproduce after clearing all site data, then browsing to a couple of articles. Eventually the site starts to show overlays, then at one point it can't be dismissed. I found that getting rid of the cookie `__tpc` seems to remove the overlays.

**Reply by stephenhawk8054** [reply]
Does it happen with uBO as well?

**Reply by stephenhawk8054** [reply]
I do see the overlay, but it's the paywall overlay for me

![image](https://github.com/user-attachments/assets/c96c664f-6c3d-488e-a4bf-3c96dcbbc73a)

**Reply by tstrmr** [reply]
It does not, and has never appeared, using standard uBO.  I wasn't even aware the site detected ad blockers.

I am aware of the subscription overlay, but that's not what I'm referring to.  You get a number of article views before it appears, and it's been there for a long time, including with standard uBO.

I'm referring to this ad blocker detected overlay:
![image](https://github.com/user-attachments/assets/4420ba4d-a75b-4d6a-895a-48eeef6b4108)

It's only appeared since I started testing uBOL in anticipation of the end of MV2.

I'm using only the  default filters - although I have tested it with all of the "Annoyances" filters turned on to see if they could get rid of the overlay, but they can't.

![image](https://github.com/user-attachments/assets/f164e53c-7665-44d3-bd76-a0f631b116d0)

**Reply by tstrmr** [reply]
Yes, it even appears on the home page:
![image](https://github.com/user-attachments/assets/7f3eb33a-e1d9-4c36-be98-c1fe576e4b50)

**Reply by stephenhawk8054** [reply]
Yeah, I mean I can only reproduce the paywall overlay, even when clearing the site cookies and data and browsing articles for a while.

**Reply by tstrmr** [reply]
Is there anything that would make my install of uBOL different from yours?  Something I could clear or change?

**Reply by gorhill** [reply]
Did you try to clear cookies and site data?

**Reply by tstrmr** [reply]
Yes - just did it again to be sure.  Also tried it with third party cookies both on and off.

I've tried this on two different computers as well with the same results.

**Reply by gorhill** [reply]
> I do see the overlay, but it's the paywall overlay for me

Yes sorry, it's the paywall overlay I can reproduce.

**Reply by tstrmr** [reply]
The paywall overlay has been consistent.  It's there after a certain number of read articles set in a post-gazette.com cookie, and is not a problem. It's been there for years, even with standard uBO.

It's the "Adblock Detected" overlay that's new since I've been using uBOL.  It renders the site pretty much useless with uBOL for me.  It never appeared with uBO.

**Reply by gorhill** [reply]
I can reproduce the anti-content blocker wall in uBO when I use this filter `@@$redirect-rule`. Filters with `redirect-rule=` option are dropped in uBOL because they can't be enforced in DNR API. Now is to find the one or more `redirect-rule=` filters which is necessary in uBOL so that we can convert those dropped filters to `redirect=` rules for that site.

---

Err ok, it's `||npttech.com/advertising.js^$script,3p,redirect-rule=noop.js`, which has been fixed with <https://github.com/uBlockOrigin/uAssets/commit/321d64217fd5767ac724431b7a1ebf8253ed0c5e>. I can see that DNR rule in the extension's package:

```json
{"action":{"redirect":{"extensionPath":"/web_accessible_resources/noop.js"},"type":"redirect"},"condition":{"domainType":"thirdParty","resourceTypes":["script"],"urlFilter":"||npttech.com/advertising.js^"},"id":416,"priority":2}
```

The only explanation left that you have the issue is that your default blocking mode is Basic. Redirects requires the default mode to be optimal or complete.

We are back at the top of the thread where I say that reporting through the chat icon is the best way to report, this way I would have seen that your default mode was at the root of the issue.

**Reply by stephenhawk8054** [reply]
> Redirects requires the default mode to be optimal or complete.

Oh I didn't notice this. So `redirect` won't happen when:

- User uses `basic` mode as default, then
- User increases the mode in a specific site to `optimal` or `complete`

?

**Reply by tstrmr** [reply]
I understand the chat icon's function now and I would have done that to begin with if I'd known what it was for.  Apologies for that.  

So even though I have set the per-site option for this site to Complete, my default mode being Basic still affects it?   

I've been leaving my default mode to Basic to minimize access to the contents of the pages I browse, but turning it up to Complete on sites that have issues. 

I wouldn't have guessed that it would work that way, honestly - that the default mode would affect the per-site outcome.   Is there no other way?

Does setting the default mode to Complete provide any access beyond what the standard uBO would?  I'd used it for years and trust it.

**Reply by gorhill** [reply]
Yes, forgot about this too, there can't be `redirect=`, `csp=`, `removeparam=` when default mode is _Basic_ or no filtering, regardless of per-site mode. The reason is that as a permission-less by default extension, it would require uBOL to explicitly request host permission for every hostname for which a request is redirected from and redirected to, for every embedded contexts which requires a csp or removeparam. uBOL only ask for permission for the current top level site.

**Reply by gorhill** [reply]
_Complete_ mode in uBOL is not worse security-wise than what uBO is doing. My general advice would be to stick to _Optimal_ as default mode and raise to _Complete_ only where needed on a per-site basis, for example when there are placeholders or sometimes inlined ads left behind.

**Reply by stephenhawk8054** [reply]
Another solution I can think of for default `basic` case is using scriptlet:

```adb
post-gazette.com##+js(aopr, setNptTechAdblockerCookie)
```

---

I just built locally with the filter and looks like it works for me.

---

Added: https://github.com/uBlockOrigin/uAssets/commit/675c584b0e973379ff94f89e036dc0a990a879f0
